src/System.Text.Json/src/System/Text/Json/JsonParseObject.cs

```diff
@@ -0,0 +1,350 @@
+                {
+                    throw new NullReferenceException();
+                }
+
+                int counter = 0;
+                for (int i = _start + RowSize; i <= _end; i += RowSize)
+                {
+                    typeCode = _buffer[i + 8];
+
+                    if (index == counter)
+                    {
+                        int newStart = i;
+                        int newEnd = i + RowSize;
+
+                        if (typeCode == (byte)JsonParser.JsonValueType.Object || typeCode == (byte)JsonParser.JsonValueType.Array)
+                        {
+                            length = BitConverter.ToInt32(_buffer, i + 4);
+                            newEnd = newEnd + RowSize * length;
+                        }
+                        return new JsonParseObject(_buffer, newStart, newEnd);
+                    }
+
+                    if (typeCode == (byte)JsonParser.JsonValueType.Object || typeCode == (byte)JsonParser.JsonValueType.Array)
+                    {
+                        length = BitConverter.ToInt32(_buffer, i + 4);
+                        i += length * RowSize;
+                    }
+
+                    counter++;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        public static explicit operator string (JsonParseObject json)
+        {
+            return GetUtf8String(json).ToString();
+        }
+
+        public static explicit operator Utf8String(JsonParseObject json)
+        {
+            return GetUtf8String(json);
+        }
+
+        private static Utf8String GetUtf8String(JsonParseObject json)
+        {
+            var typeCode = json._buffer[json._start + 8];
+
+            if (typeCode == (byte)JsonParser.JsonValueType.Object || typeCode == (byte)JsonParser.JsonValueType.Array)
+            {
+                throw new InvalidCastException();
+            }
+
+            int location = BitConverter.ToInt32(json._buffer, json._start);
+            int length = BitConverter.ToInt32(json._buffer, json._start + 4);
+            return new Utf8String(json._buffer, location, length);
+        }
+
+        public static explicit operator bool (JsonParseObject json)
+        {
+            var typeCode = json._buffer[json._start + 8];
+
+            if (typeCode == (byte)JsonParser.JsonValueType.Object || typeCode == (byte)JsonParser.JsonValueType.Array)
+            {
+                throw new InvalidCastException();
+            }
+
+            int length = BitConverter.ToInt32(json._buffer, json._start + 4);
+            if (length < 4 || length > 5)
+            {
+                throw new InvalidCastException();
+            }
+
+            int location = BitConverter.ToInt32(json._buffer, json._start);
+            bool isTrue = json._buffer[location] == 't' && json._buffer[location + 1] == 'r' && json._buffer[location + 2] == 'u' && json._buffer[location + 3] == 'e';
```

Comment:
Yes, agreed. Although, then I should also check that length is actually 4. Currently, if json buffer had truex, it would still pass isTrue == true (since the first 4 characters match).